### PR TITLE
Added pre-commit hooks and GitHub Action to run `black` for formatting code.

### PR DESCRIPTION
**Pull request by maouw**
- Requires an upgrade to the oldest version of Python that is available on `klone`  (`python3.9`) in order to run precommit hooks.
- Adds a `.gitignore` to exclude irrelevant files in Git
- Adds configuration for `flake8`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,6 +46,16 @@ python3 -m pip install --user .
 
 If successful, then `hyakvnc` should be installed to `~/.local/bin/`.
 
+#### Optional dependencies for development
+
+The optional dependency group`[dev]` in  `pyroject.toml` includes dependencies useful for development, including [pre-commit](https://pre-commit.com/) hooks that run in order to commit to the `git` repository.
```
Comment (arokem) [inline README.md:51]:
```suggestion
The optional dependency group`[dev]` in  `pyproject.toml` includes dependencies useful for development, including [pre-commit](https://pre-commit.com/) hooks that run in order to commit to the `git` repository.
```

**Review by arokem — APPROVED** [review]
LGTM!

File: README.md (hunk 1 above)
Comment (maouw) [reply README.md]: Fixed the typo. Also formatted README to be compliant with [`markdownlint`](https://github.com/DavidAnson/markdownlint).

**Review by maouw — COMMENTED** [review]
Fixed typo.

**Pull request closed without merging**